ApiDocs.Validation/OData/ExtensionMethods.cs

```diff
@@ -51,23 +66,229 @@ internal static string AttributeValue(this XElement xml, XName attributeName)
+                                      select et;
+            if (matchingEntityType.Any())
+                return matchingEntityType.First();
+
+            // Look up actions
+            var matchingAction = (from et in schema.Actions where et.Name == typeName select et);
+            if (matchingAction.Any())
+                return matchingAction.First();
 
-            var schema = (from s in schemas where s.Namespace == schemaName select s).FirstOrDefault();
-            if (null == schema) return null;
+            // Look up functions
+            var matchingFunctions = (from et in schema.Functions where et.Name == typeName select et);
+            if (matchingFunctions.Any())
+                return matchingFunctions.First();
 
-            var matchingComplexType = (from ct in schema.ComplexTypes where ct.Name == typeName select ct).FirstOrDefault();
-            if (null != matchingComplexType) return matchingComplexType;
+            return null;
+        }
+
+        public static T ResourceWithIdentifier<T>(this IEnumerable<Schema> schemas, string identifier)
+        {
+            var type = schemas.FindTypeWithIdentifier(identifier);
+            if (type != null && type is T)
+            {
+                return (T)type;
+            }
+
+            throw new KeyNotFoundException("Unable to find type identifier '" + identifier + "' as '" + typeof(T).Name + "'.");
+        }
+
+        /// <summary>
+        /// Look up a type in the EntityFramework based on the type identifier.
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="edmx"></param>
+        /// <param name="identifier"></param>
+        /// <returns></returns>
+        public static T ResourceWithIdentifier<T>(this EntityFramework edmx, string identifier)
+        {
+            return edmx.DataServices.Schemas.ResourceWithIdentifier<T>(identifier);
+        }
+
+        internal static IODataNavigable LookupNavigableType(this EntityFramework edmx, string identifier)
+        {
+            var foundType = edmx.DataServices.Schemas.FindTypeWithIdentifier(identifier);
+            if (null != foundType)
+            {
+                return (IODataNavigable)foundType;
+            }
+
+            SimpleDataType simpleType = identifier.ToODataSimpleType();
+            if (simpleType != SimpleDataType.Object)
+                return new ODataSimpleType(simpleType);
+
+            throw new InvalidOperationException("Could not resolve type identifier: " + identifier);
+        }
+
+        /// <summary>
+        /// Resolve an IODataNavigable instance into a type identifier string.
+        /// </summary>
+        /// <param name="edmx"></param>
+        /// <param name="type"></param>
+        /// <returns></returns>
+        public static string LookupIdentifierForType(this EntityFramework edmx, IODataNavigable type)
+        {
+            if (type is ODataSimpleType)
```

Comment:
Yeah, that's a good point. There's data in the table that describes better the true format of the property. We should probably combine the two together to make sure that the type in the example matches the type defined in the table.